From c0686879ac66c1933aefbb62b69afb0c9a0db912 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Mon, 9 Sep 2013 17:13:15 +0200
Subject: [PATCH] patch: pdf-no-high-bpp

Squashed commit of the following:

commit 9f7d97e114389595481f6e9d3ac1038972f3f73b
Author: Nils Philippsen <nils@redhat.com>
Date:   Mon Sep 9 17:08:38 2013 +0200

    avoid producing PDFs with bpp > 8
---
 src/xsane-save.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/xsane-save.c b/src/xsane-save.c
index 5461bf1..75e0a63 100644
--- a/src/xsane-save.c
+++ b/src/xsane-save.c
@@ -4205,6 +4205,18 @@ int xsane_save_pdf(FILE *outfile, FILE *imagefile, Image_info *image_info, float
 
   *cancel_save = 0;
 
+  if (image_info->depth > 8)
+  {
+    char buf[TEXTBUFSIZE];
+
+    snprintf(buf, sizeof(buf), "%s %s", ERR_DURING_SAVE, "PDF doesn't allow bit depths > 8");
+    DBG(DBG_error, "%s\n", buf);
+    xsane_back_gtk_decision(ERR_HEADER_ERROR, (gchar **) error_xpm, buf, BUTTON_OK, NULL, TRUE /* wait */);
+    *cancel_save = 1;
+
+    goto bail_out;
+  }
+
   xsane_save_pdf_create_document_header(outfile, &xref, 1, flatedecode);
 
   if (apply_ICM_profile && (cms_function == XSANE_CMS_FUNCTION_EMBED_SCANNER_ICM_PROFILE))
@@ -4232,6 +4244,8 @@ int xsane_save_pdf(FILE *outfile, FILE *imagefile, Image_info *image_info, float
     *cancel_save = 1;
   }
 
+bail_out:
+
  return (*cancel_save);
 }
 
-- 
1.8.3.1

